ckanext/data_qld/plugin.py

```diff
@@ -191,13 +198,54 @@ def get_blueprint(self):
 
     # IQA
     def custom_resource_score(self, resource, resource_score):
+        resource_score_format = resource_score.get('format').upper() if resource_score.get('format') is not None else ''
+        resource_format = resource.format.upper() if resource.format is not None else ''
         # If resource openness_score is 3 and format is CSV
-        if resource_score.get('openness_score', 0) == 3 and resource_score.get('format', '').upper() == 'CSV':
+        if resource_score.get('openness_score', 0) == 3 and resource_score_format == 'CSV':
             # If resource has a JSON schema which validated successfully, set score to 4
-            if hasattr(resource, 'extras') and resource.extras.get('schema', None) and resource.extras.get('validation_status', None) == 'success':
+            if hasattr(resource, 'extras') and resource.extras.get('schema', None) and resource.extras.get(
+                    'validation_status', None) == 'success':
                 resource_score['openness_score'] = 4
-                resource_score['openness_score_reason'] = toolkit._('Content of file appeared to be format "{0}" which receives openness score: {1}.'
-                                                                    .format(resource_score.get('format', ''), resource_score.get('openness_score', '')))
+                resource_score['openness_score_reason'] = toolkit._(
+                    'Content of file appeared to be format "{0}" which receives openness score: {1}.'
+                    .format(resource_score_format, resource_score.get('openness_score', '')))
+
+        # QA cannot determine file formats that are not part of its own
+        # 'resource_format_openness_scores.json' file and CKAN resource_formats.json file
+        # The below are dataqld specific file formats that are not part of the default
+        # CKAN resource_formats.json file and need custom scoring
+
+        # If QA believes the resource is a TIFF file, check the resource format selected,
+        # if it's GEOTIFF apply custom score
+        if resource_score_format == 'TIFF' and resource_format == 'GEOTIFF':
+            resource_score['openness_score'] = resource_score['openness_score'] = qa_lib.resource_format_scores().get(
+                resource_format)
+            resource_score['openness_score_reason'] = toolkit._(
+                'Content of file appeared to be format "{0}" which receives openness score: {1}.'
+                .format(resource_format, resource_score.get('openness_score', '')))
+
+        # If QA believes the resource is a ZIP file, check the resource format selected,
```

Comment:
The QA sniffing already has special logic for inspecting the contents of zipfiles. Is that being taken into account?

Reply:
@ThrawnCA these are overrides for the Data.Qld ruleset - implemented here to avoid customising the ckanext-qa extension